internal/clientcache/internal/cache/refresh.go

```diff
@@ -124,3 +169,56 @@ func (r *RefreshService) Refresh(ctx context.Context, opt ...Option) error {
 	}
 	return retErr
 }
+
+// FullFetch is used for users of boundary instances which do not support
+// refresh tokens. For these users a call to FullFetch retrieves all resources
+// for that user and updates the database with the result. Users are identified
+// as being in this group if they have at least one resource in the cache and
+// do not have any refresh tokens. Users who do not have any resources nor
+// refresh tokens are updated using Refresh since a query for no resources
```

Comment:
This is a bit confusing.  "a query for no resources should be just as quick as q
query using refresh tokens".  It sounds like your comparing a query that returns
no resources but I think you're asserting that a full refresh costs the same as
a query using refresh tokens when you're starting with no resources in the cache.